src/main/java/de/embl/cba/mobie/ui/viewer/SourcesPanel.java

```diff
@@ -291,17 +290,21 @@ public void addSourceToPanelAndViewer( String sourceName )
         }
         else
         {
-            final SourceAndMetadata< ? > sam = getSourceAndMetadata( sourceName );
-            sourceNameToSourceAndMetadata.put( sam.metadata().displayName, sam );
+            final SourceAndMetadata< ? > sam = getSourceAndDefaultMetadata( sourceName );
```

Comment:
...and here you go through some getter method:

`getSourceAndDefaultMetadata(...`

Is there a reason? If not, maybe do the same thing in both cases?

Reply:
This is due to the default metadata vs current metadata. In the setColor call it needs to get the current metadata (i.e. maybe the bookmark metadata that was used to create it). In the addSourceToPanelAndViewer, it needs to get the actual default data so that bookmarks don't persist etc... I'll rename sourceNameToSourceAndMetadata to make this clearer